src/Model/lbsProcessor.py

```diff
@@ -257,14 +258,18 @@ def compute_cmf_underloads(self, l_ave, pmf_type=0):
 
         # Distinguish between different PMF types
         if not pmf_type:
-            # Initialize ancillary values
+            # Determine whether one underloaded is actually overloaded
+            loads = self.known_underloads.values()
+            l_max = max(loads)
+            factor = l_max if l_max > l_ave else l_ave
```

Comment:
Computing `factor` this way will lead to a zero contribution to the CMF for the highest loaded processor being considered, right?  If there is only one that becomes overloaded, we won't be able to select it for more transfers.  It will definitely change the CMF values for the other processors, however, making it slightly more likely that we choose an underloaded with higher load vs lower load.

Reply:
We could potentially calculate `factor` using the load of the truly overloaded processor that is computing the CMF.  Due to making it more likely to choose less severely underloaded processors, however, I'm not sure if it would have the desired effect.

Reply:
@nlslatt thanks for the review & comments

By **the** truly OL processor, do you mean the one that is the most overloaded (amongst all those that were previously underloaded prior to a first round of object transfers) across all ranks? If so, making its load known to all in order to compute the CMF with the same normalizing factor would require some global communication as not all ranks would necessarily about it otherwise. 
But indeed in the approach above the normalization is not done in the same way across all ranks (those who are not aware of "overloaded-underloaded" targets simply divide by the average load).

Reply:
@ppebay No, by the truly overloaded processor, I meant the processor computing the CMF, i.e., the one that is considering offloading work to a less-loaded processor.  Because the modified grapevine criterion involves not letting the selected processor become more overloaded than the one offloading the work, we know that the offloading processor will not make any processor included in the CMF calculation become more overloaded than itself.  Instead of taking `factor` from the maximum load included in the CMF calculation, we could take it from the (necessarily greater) load of the offloading processor.  We would still avoid negatives in the CMF, and would have non-zero CMF contributions from all included processors.  Whether doing so would give us extra wiggle room or lead us down the wrong path isn't immediately clear to me.

Reply:
@nlslatt 
Regarding your first comment about `factor` I agree with what you are describing, but 2 things to be considered here:
- it is similar to what would happen for a processor whose load is exactly at the average (and if no other rank has become overloaded): it would not get chosen. We could alleviate that by multiplying the normalization factor by 0.9 or 0.99, which I have tried but does not yield noticeable effects either way;
- only **the** most overloaded rank, amongst all known "underloaded" processors known to the sender, will have a 0 contribution to the CMF and therefore not be selected **at that iteration**, so
  -  in subsequent iterations it might no longer be the most overloaded rank in the list, and therefore become "actionable" again ;
  - and if several "known underloaded" have become overloaded, only the ones with maximal load will have that 0 contribution. Other, "less overloaded" ranks would have a non-0 contribution (as their L_max:L ratios would be smaller than 1) and therefore would remain "selectable".

Reply:
@nlslatt understood now regarding the max load to be used for the CMF. That's an interesting angle. I shall experiment with it a little. Pushing a change to the code now BTW

Reply:
@nlslatt Nicole I looked into histograms given by your proposed variant, based on using the sender's load as the denominator in the CMF computation. It is indeed giving slightly different results, but it's hard to tell whether on average they end up with better or worse load imbalances. At that stage I think it's an interesting option that deserves further consideration, so I've put it in the LBAF code as a `pmf_type` which can be either 0 (the fixed version using the max underload when it's over the average) or 1 (the "NS variant"). You can flip the switch in line 297 of `lbsRuntime.py`, right now this requires editing that line, but we should probably make this a command line option for more systematic investigation.

In any event I think we should also experiment with it on the `GossipLB` side.

Reply:
@lifflander @nlslatt can we resolve this thread and merge? Thanks